inst/tests/tests.Rraw

```diff
@@ -18111,3 +18111,11 @@ test(2238.9, NA %notin% c(1:5, NA), FALSE)
 
 # shift actionable error on matrix input #5287
 test(2239.1, shift(matrix(1:10, ncol = 1)), error="consider wrapping")
+
+# names<- retains index and key, #5125, #5126, #5126, #5128
```

Comment:
Great work.
Could we add rest of unit tests added in #5133?
If some of them fails then possibly comment them out with explanation. So we have clarity what from #5133 is being covered.

Reply:
good idea. I skipped originally bc I wasn't sure whether output change tests are worth copying.

I'll read more carefully and copy over the worthwhile tests.

Reply:
OK, copied over the tests. They caught two things:

 1. `colnames(x) <-` is now treated the same a `names(x) <-`. Requires a new S3 method registration.
 2. Doing `names(x) <- "..."; set(x, j=j, val=val)` is fixed with a simple `setalloccol()`. Maybe not ideal, but a good expedient.

Reply:
I don't see any tests being commented out. Does it mean we cover all tests from Matt's PR?

Reply:
I didn't copy over anything where the only diff is in error/message output, i.e.

https://github.com/Rdatatable/data.table/pull/5133/files#diff-e3243f3780ce7d303c3317f73945310bfc37e45d193568246246aca20e3270aeR1336-R1341

https://github.com/Rdatatable/data.table/pull/5133/files#diff-e3243f3780ce7d303c3317f73945310bfc37e45d193568246246aca20e3270aeR1354-R1359

The new tests of actual code are copied over and pass:

https://github.com/Rdatatable/data.table/pull/5133/files#diff-e3243f3780ce7d303c3317f73945310bfc37e45d193568246246aca20e3270aeR1532-R1552